File: Python/grep/grep.py
#!/usr/bin/env python

import sys

#TODO: Grep stdin
#TODO: Grep from a file

def main():
    
    arg = sys.argv[1]
    
    stdin = sys.stdin.read()
    tokenizedInput = stdin.split('\n')
    for line in tokenizedInput:
        if arg in line:
            sys.stdout.write(line + '\n')
        else:
            pass

    sys.exit(0)

if __name__ == "__main__":
    main()

File: Node.js/grep/grep.js
process.stdin.setEncoding('utf8')

let lines;
let grep_string;

grep_string = process.argv[2]

process.stdin.on('readable', () => {
    const chunk = process.stdin.read();
    if (chunk !== null) {
        lines = chunk.split('\n');
        for (i in lines) {
            if (lines[i].includes(grep_string) == true) {
                process.stdout.write(`${lines[i]}\n`);
            }
        }
    }
})

